R/pr.R

```diff
@@ -0,0 +1,137 @@
+pr_create <- function(branch) {
+  check_uses_github()
+
+  cur_branch <- proj_git_branch_get()
+  if (cur_branch$name == branch) {
+    return(invisible(TRUE))
+  }
+
+  pr_branch <- proj_git_branch_find(branch)
+  if (is.null(pr_branch)) {
+    if (cur_branch$name != "master") {
+      if (nope("Start pull-request from non-master branch?")) {
+        return(invisible(FALSE))
+      }
+    }
+
+    done("Creating branch {value(branch)}")
+    head <- git2r::commits(repo, n = 1)[[1]]
+    pr_branch <- git2r::branch_create(head, branch)
+  }
+
+  done("Switching to branch {value(branch)}")
+  git2r::checkout(pr_branch)
+
+  invisible()
+}
+
+pr_push <- function(force = FALSE) {
+  check_on_branch()
+  check_uses_github()
+  check_uncommitted_changes()
+
+  done("Pushing changes to GitHub PR")
+  git2r::push(
+    object = proj_git_repo(),
+    name = "origin",
+    refspec = paste0("refs/heads/", cur_branch$name),
+    force = force
+  )
+}
+
+pr_pull <- function() {
+  check_on_branch()
+  check_uses_github()
+  check_uncommitted_changes()
+
+  done("Pulling changes from GitHub PR")
+  git2r::pull(repo)
+}
+
+pr_update <- function() {
+  done("Fetching master branch from GitHub")
+  git2r::fetch(repo, "origin", refspec = "refs/head/master")
+
+  done("Merging master branch")
+  out <- merge(repo, "origin/master")
```

Comment:
I think I rebase as or more often as I merge. I guess it doesn't matter that much given our "squash and merge" mentality.

Reply:
Reasons why I merge here:

* I couldn't figure out how to rebase with git2r

* Rebase will mean that you have to force push, which makes me a little worried now that others might contributing to the PR branch (esp. via suggestions)